android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -402,13 +402,18 @@ protected void onAttachedToWindow() {
             }
             InitBraveNewsController();
             initNews();
-
             if (BraveActivity.getBraveActivity() != null && mIsNewsOn) {
                 new Handler().post(() -> {
                     Tab tab = BraveActivity.getBraveActivity().getActivityTab();
-
                     if (tab != null && tab.getUrl().getSpec() != null
                             && UrlUtilities.isNTPUrl(tab.getUrl().getSpec())) {
+                        // purges display ads on tab change
+                        if (BraveActivity.getBraveActivity().getLastTabId() != tab.getId()) {
+                            mBraveNewsController.onDisplayAdPurgeOrphanedEvents();
+                        }
```

Comment:
Does this cover if a user has n tabs, and restarts the browser?

Reply:
Yes. on restart the tab id gets incremented and it will be different